WebFileSystem.ts

```diff
@@ -481,6 +481,15 @@ class WebFileSystem extends webdav.FileSystem {
     async createResource (path: Path, user: User, type: webdav.ResourceType) : Promise<Error> {
         // TODO: Handle permissions
 
+        // checks if file already exists and if filename contains bad characters (e.g. "§%?&....")
+        if (this.resources.get(user.uid).has(path.toString())) {
+            logger.info(`Resource ${path} already exists.`)            
+            return webdav.Errors.ResourceAlreadyExists
+        } else if (path.fileName() !== encodeURI(path.fileName())){
+            logger.info(`Resourcename ${path.fileName()} not allowed.`)
+            return new Error('The Name of the Resource contains forbidden characters')
```

Comment:
This leads to an 500 Error (Internal Server Error), which doesn't look good for the user. Probably should return webdav-Error.

Reply:
I am not 100% sure, but this could be because of the HTTP Codes stated in this File: https://github.com/OpenMarshal/npm-WebDAV-Server/blob/master/src/server/HTTPCodes.ts
Apparently `Illegal Arguments` isn't part of this list and therefore it returns the status code 500.